### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\ufeff'

### ISSUE THREAD

**Issue by azenakhi** [ansible/ansible#13765]
Hello,
I wanted to test ansible-2.0.0. 
When I execute the command "ansible all -m ping", I have the following output. 

```
ansible 2.0.0
  config file = ansible2/conf/ansible.cfg
  configured module search path = ansible-2.0.0/library
No config file found; using defaults
Loaded callback minimal of type stdout, v2.0
<ï»¿GUEST01> ESTABLISH SSH CONNECTION FOR USER: None
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "ansible2/lib/ansible/executor/process/worker.py", line 114, in run
    self._shared_loader_obj,
  File "ansible2/lib/ansible/executor/task_executor.py", line 119, in run
    res = self._execute()
  File "ansible2/lib/ansible/executor/task_executor.py", line 401, in _execute
    result = self._handler.run(task_vars=variables)
  File "ansible2/lib/ansible/plugins/action/normal.py", line 30, in run
    results.update(self._execute_module(tmp=tmp, task_vars=task_vars))
  File "ansible2/lib/ansible/plugins/action/__init__.py", line 405, in _execute_module
    tmp = self._make_tmp_path()
  File "ansible2/lib/ansible/plugins/action/__init__.py", line 211, in _make_tmp_path
    result = self._low_level_execute_command(cmd, sudoable=False)
  File "ansible2/lib/ansible/plugins/action/__init__.py", line 520, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "ansible2/lib/ansible/plugins/connection/ssh.py", line 603, in exec_command
    raise e
UnicodeEncodeError: 'ascii' codec can't encode character u'\ufeff' in position 309: ordinal not in range(128)
                                                                                                                                                                                     ï»¿
GUEST01| FAILED! => {
    "failed": true,
    "stdout": ""
```

Best regards.

**Comment by abadger** [comment]
I believe this problem is fixed in 2.0.0-rc4.  The line number and code that I'm seeing in the traceback were changed in rc4 to fix UnicodeError tracebacks such as you're seeing.  In rc4 it looks like this: https://github.com/ansible/ansible/blob/stable-2.0/lib/ansible/plugins/connection/ssh.py#L610

If you could give rc4 a try from: http://releases.ansible.com/ansible/  I think that you'll find that this problem has been solved.

If not, we'll need to know some more information.  UnicodeErrors occur when non-ascii values are fed into ansible and then not handled correctly.  In order to duplicate and solve the issues we need to know where those non-ascii values are entering the system.  (For instance, through a hostname, usename, environment variable, or playbook.)

**Comment by abadger** [comment]
Going to assume this was fixed per the analysis above.
